### PR TITLE
CVE-2013-1965 OGNL injection through parameter names

### PR DESCRIPTION
**Pull request by renchauret**

### CODE REVIEW

File: cves/CVE-2013-1965.yml [hunk 1]
```diff
@@ -45,7 +45,12 @@ description_instructions: |
   that outsiders to Struts would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description:
+description: |
```
Comment (MattThyng) [inline cves/CVE-2013-1965.yml:48]: What is OGNL? A brief explanation may be nice.

Comment (renchauret) [reply cves/CVE-2013-1965.yml:48]: OGNL is a simplified expression language for Java.  I can add a bit more of a thorough description to the yaml

File: cves/CVE-2013-1965.yml [hunk 2]
```diff
@@ -162,8 +176,11 @@ lessons:
     If you think of another lesson we covered in class that applies here, feel
     free to give it a small name and add one in the same format as these.
   defense_in_depth:
-    applies:
-    note:
+    applies: true
+    note: |
```
Comment (randallkentbrown) [inline cves/CVE-2013-1965.yml:181]: So the vulnerability is still applicable if this feature is enabled?

File: cves/CVE-2013-1965.yml [hunk 3]
```diff
@@ -205,4 +227,13 @@ mistakes:
...
-  answer:
+  answer: |
+    The problem which caused this vulnerability was one of design. The conclusion 
+    eventually reached was that OGNL evaluation in XWork core simply cannot easily 
+    be implemented in a secure manner. After multiple attempts to sanitize input, 
+    the solution was to just disable OGNL evaluation by default.
+    
+    Additionally, the multiple attempts at fixes show that testing may not have 
+    been thorough enough to push those fixes to production. New edge cases kept 
+    popping up and requiring small adjustments  until this final solution was
```
Comment (randallkentbrown) [inline cves/CVE-2013-1965.yml]: Minor, but there are two spaces here in "... adjustments  until ..."

File: cves/CVE-2013-1965.yml (hunk 2 above)
Comment (renchauret) [reply cves/CVE-2013-1965.yml:181]: That is correct:  The fix was to disable this feature and give a warning if another module tries to enable it

Comment (randallkentbrown) [reply cves/CVE-2013-1965.yml:181]: Gotcha. I would add a blurb to the description that mentions that, but that's something you can easily get from context on the fix

File: cves/CVE-2013-1965.yml (hunk 1 above)
Comment (based-avocado) [reply cves/CVE-2013-1965.yml:48]: Agreeing with Matt here. A little bit of an explanation on OGNL would be convenient

File: cves/CVE-2013-1965.yml [hunk 4]
```diff
@@ -205,4 +227,13 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those in the software
     engineering industry would find interesting.
-  answer:
+  answer: |
+    The problem which caused this vulnerability was one of design. The conclusion 
+    eventually reached was that OGNL evaluation in XWork core simply cannot easily
```
Comment (based-avocado) [inline cves/CVE-2013-1965.yml:233]: Why were they initially performing OGNL evaluation? Might be nice to include a little context about why they opted to use this technology initially in this answer, if you can find it

**Comment by based-avocado** [comment]
Upvotes: 2

File: cves/CVE-2013-1965.yml [hunk 5]
```diff
@@ -103,9 +110,11 @@ unit_tested:
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
     Must be just "true" or "false".
-  answer:
```
Comment (andymeneely) [inline cves/CVE-2013-1965.yml:106]: +5 upvotes from me

File: cves/CVE-2013-1965.yml (hunk 4 above)
Comment (renchauret) [reply cves/CVE-2013-1965.yml:233]: Originally, XWork core was a separate repository, and OGNL evaluation was done there.  This vulnerability was migrated with XWork core to this repository, and history predating that migration seems to be mostly wiped clean.

**Comment by randallkentbrown** [comment]
3 upvotes

**Pull request merged**